epochtalk/epochtalk
# Redis

Exchange memdown for redis, Image expiration on DB,  and post - image reference

# Dependencies:
Redis must be installed and running
The default port and host are fine

# Testing procedures:
## Testing redis logins
* Log into server
* Without closing the browser tab, restart the server
* Refresh page and ensure you are still logged in

## Testing image expirations
* Set the default image expiration to something low (15 secs)
* Open the post editor and upload an image
* Ensure that the image is in the image_expirations table, and on the cdn
* Wait till the image expires
* Ensure that the image is no longer in the image_expirations table and the image is now gone

## Testing image references
* Open the post editor and upload an image, save this post
* Ensure that there is a new row in the images_posts table
* Purge the post
* Ensure that the table row has a null post_id
* Wait for the image interval (same time as above)
* Ensure that the entry in the table is gone along with the image on cdn.

## Testing image deletion on no more references
* Do the same as above but update the image (either new or no image)
* Ensure the images_posts table has at least one row with null post_id and one new entry
* Wait for the image interval (same time as above)
* Ensure that there is only one entry now and the image is deleted from cdn.
